woocommerce-services.php

```diff
@@ -528,6 +528,175 @@ public function init_core_wizard_payments_config() {
+					$settings_meta['api_credentials']['description'] .= '<br /><br />' . $api_creds_text;
+				}
+				if ( empty( $settings['sandbox_api_username'] ) ) {
+					$api_creds_text = sprintf( $api_creds_template, add_query_arg( 'environment', 'sandbox', $reset_link ) );
+					$settings_meta['sandbox_api_credentials']['description'] .= '<br /><br />' . $api_creds_text;
+				}
+			}
+
+			return $settings_meta;
+		}
+
+		public function paypal_ec_maybe_set_reroute_requests() {
+			if (
+				empty( $_GET['reroute_requests'] ) ||
+				empty( $_GET['nonce'] ) || ! wp_verify_nonce( $_GET['nonce'], 'reroute_requests' )
+			) {
+				return;
+			}
+
+			$ppec_settings = get_option( 'woocommerce_ppec_paypal_settings', array() );
+			$ppec_settings['reroute_requests'] = 'yes' === $_GET['reroute_requests'] ? 'yes' : 'no';
+			if ( isset( $_GET['environment'] ) ) {
+				$ppec_settings['environment'] = 'sandbox' === $_GET['environment'] ? 'sandbox' : 'live';
+			}
+			update_option( 'woocommerce_ppec_paypal_settings', $ppec_settings );
+
+			wp_safe_redirect( wc_gateway_ppec()->get_admin_setting_link() );
+		}
+
+		public function paypal_ec_settings( $settings ) {
+			$settings['paymentaction'] = 'sale';
+
+			if ( empty( $settings['api_username'] ) ) {
+				$settings['api_username'] = '';
+			}
+			if ( empty( $settings['sandbox_api_username'] ) ) {
+				$settings['sandbox_api_username'] = '';
+			}
+
+			return $settings;
+		}
+
+		/**
+		 * Modify PPEC plugin behavior to facilitate proxying and authenticating requests via server
+		 */
+		public function paypal_ec_setup() {
+			$ppec_settings = get_option( 'woocommerce_ppec_paypal_settings', array() );
+			add_filter( 'woocommerce_paypal_express_checkout_settings', array( $this, 'paypal_ec_settings_meta' ) );
+			add_action( 'load-woocommerce_page_wc-settings', array( $this, 'paypal_ec_maybe_set_reroute_requests' ) );
+
+			if ( isset( $ppec_settings['reroute_requests'] ) && 'yes' === $ppec_settings['reroute_requests'] ) {
+				// If empty, populate Sandbox API Subject with Live API Subject value
+				if (
+					! isset( $ppec_settings['sandbox_api_subject'] ) &&
+					! isset( $ppec_settings['sandbox_api_username'] ) &&
+					! isset( $ppec_settings['api_username'] )
+				) {
+					$ppec_settings['sandbox_api_subject'] = $ppec_settings['api_subject'];
+					update_option( 'woocommerce_ppec_paypal_settings', $ppec_settings );
+				}
+
+				$username_key = 'sandbox' === $ppec_settings[ 'environment' ] ? 'sandbox_api_username' : 'api_username';
+				if ( empty( $ppec_settings[ $username_key ] ) ) {
+					// Reroute requests from the PPEC extension via WCS to pick up API credentials
+					add_filter( 'woocommerce_paypal_express_checkout_request_endpoint', array( $this, 'paypal_ec_endpoint' ) );
+
+					add_filter( 'woocommerce_payment_gateway_supports', array( $this, 'paypal_ec_supports' ), 10, 3 );
+					add_filter( 'option_wc_gateway_ppce_prompt_to_connect', array( $this, 'paypal_ec_prompt_to_connect' ) );
```

Comment:
I suspect this will change given our discussion on using a new custom banner, but:

Using `pre_option_{option}` here would be better since you don't need the existing option value.

Reply:
That was working from me but I shied away after spotting this in [documentation](https://developer.wordpress.org/reference/hooks/pre_option_option/):
> Passing a truthy value to the filter will short-circuit retrieving the option value, returning the passed value instead.

In this case, I want a falsy value in the case of suppressing the notice. Is `pre_option_{option}` safe to use?